app/controllers/api/v1/data_studio/school_rollout_controller.rb

```diff
@@ -7,70 +7,61 @@ class SchoolRolloutController < Api::ApiController
+          # instantiates School and associated objects and then serializes all of them in a similar
+          # fashion to the other API endpoints.
           query = <<~SQL
-            SELECT
-              schools.id,
-              schools.name,
-              schools.urn,
-              nomination_emails.sent_at,
-              nomination_emails.opened_at,
-              nomination_emails.notify_status,
-              (icp.id IS NOT NULL) AS induction_tutor_nominated,
-              icp.created_at AS tutor_nominated_time,
-              (users.current_sign_in_at IS NOT NULL) AS induction_tutor_signed_in,
-              school_cohorts.induction_programme_choice,
-              school_cohorts.created_at AS programme_chosen_time,
-              (partnerships.id IS NOT NULL) AS in_partnership,
-              partnerships.created_at AS partnership_time,
-              partnerships.challenge_reason AS partnership_challenge_reason,
-              partnerships.challenged_at AS partnership_challenge_time,
-              lead_providers.name AS lead_provider,
-              delivery_partners.name AS delivery_partner,
-              cip.name AS chosen_cip,
-              school_cohorts.updated_at AS cip_chosen_time 
-            FROM
-              schools
-              FULL OUTER JOIN nomination_emails ON schools.id = nomination_emails.school_id
+            SELECT json_build_object('data',array_agg(query_rows))::text FROM (
+              SELECT
+                schools.id AS id,
+                'school_rollout' AS type,
+                json_build_object(
+                  'name', schools.name,
+                  'urn',  schools.urn,
+                  'sent_at', nomination_emails.sent_at,
+                  'opened_at', nomination_emails2.opened_at,
+                  'notify_status', nomination_emails.notify_status,
+                  'induction_tutor_nominated', (icp.id IS NOT NULL),
+                  'tutor_nominated_time', icp.created_at,
+                  'induction_tutor_signed_in', (users.current_sign_in_at IS NOT NULL),
+                  'induction_programme_choice', school_cohorts.induction_programme_choice,
+                  'programme_chosen_time', school_cohorts.created_at,
+                  'in_partnership', (partnerships.id IS NOT NULL),
+                  'partnership_time', partnerships.created_at,
+                  'partnership_challenge_reason', partnerships.challenge_reason,
+                  'partnership_challenge_time', partnerships.challenged_at,
+                  'lead_provider', lead_providers.name,
+                  'delivery_partner', delivery_partners.name
+                ) as attributes
+              FROM schools
+              LEFT OUTER JOIN nomination_emails ON schools.id = nomination_emails.school_id
+              LEFT OUTER JOIN nomination_emails nomination_emails2 ON schools.id = nomination_emails2.school_id
```

Comment:
What is `nomination_emails2` in here? As in, how is it different from regular `nomination_emails`?

Reply:
It looks to me as if the query needed to be split out so that details of the last `sent_at` and `opened_at` could be returned independently i.e. they might not be related to the same nomination email for the school.